### DISCUSSION TITLE
wire:click is not firering anything, Laravel-Jetstream-Livewire-Tailwindcss

### DISCUSSION THREAD

**Discussion by kamdjouduplex** [livewire/livewire#3911] [1. Help] [answered]
I have been strugle to make the wire:click work on my Laravel jetstream app.
I added jetstream using the livewire stack on my laravel project and the authentication is working well.

I first added a component using `php artisan make:component AddSection` now in the view of my component, I added this:
**views/components/add-section.blade.php**
```
<div>
    <button wire:click="addsection()" class="inline-flex items-center px-4 py-2 border-4 border-green-500 font-bold text-xs text-green-500 uppercase tracking-widest hover:bg-gray-700 active:bg-gray-900 focus:outline-none focus:border-gray-900 focus:ring focus:ring-gray-300 disabled:opacity-25 transition my-4">
        <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 4v16m8-8H4" />
        </svg>
        Nouvelle Section
    </button>
</div>
```
and in my Component I added this:
**App/Http/View/Components**
    
```
<?php

namespace App\View\Components;

use Illuminate\View\Component;

class AddSection extends Component
{
    /**
     * Create a new component instance.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }   

    public function addsection(){
        dd('Good Good');
    }


    /**
     * Get the view / contents that represent the component.
     *
     * @return \Illuminate\Contracts\View\View|\Closure|string
     */
    public function render()
    {
        return view('components.add-section');
    }
}
```

In the parent view I am extending a custom layout I created like this:
**views/parent-folder/parent.blabe.php**
```
<x-base-layout>
    <div class="max-w-7xl mx-auto bg-white">
        <div class="p-8">
            <h3 class="font-bold text-2xl">Editer le programme</h3>
        </div>
        <hr>
        <div class="p-8" id="coursSectionWraper">
            {{-- <x-add-section />  or the line below none works --}}  
            <x-add-section></x-add-section>
        </div>
    </div>
</x-base-layout>
```
My custom layout look like this:
**views/layouts/base.blabe.php**
<div>

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>{{ config('app.name', 'Laravel') }}</title>

    <!-- Fonts -->
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Azeret+Mono:wght@100;200;400;600;800&display=swap" rel="stylesheet">

    <!-- Styles -->
    <link rel="stylesheet" href="{{ mix('css/app.css') }}">

    @livewireStyles
    <!-- Scripts -->
    <script src="{{ mix('js/app.js') }}" defer></script>
</head>

<body class="font-sans antialiased">
    <x-jet-banner />

    <div class="min-h-screen bg-gray-100">
        @livewire('navigation-menu')

        <!-- Page Content -->
        <main>
            <div class="grid grid-cols-12">
                <div class="col-span-2">
                    <x-sidebar />
                </div>
                <div class="col-span-10">
                    {{ $slot }}
                </div>
            </div>
        </main>
    </div>

    @stack('modals')

    @livewireScripts
    <script src="{{ asset('js/vendor/ckeditor.js') }}"></script>
    <script>
        ClassicEditor.create(document.querySelector('#editor'), {
                toolbar: ['bold', 'italic', 'undo', 'redo', 'numberedList', 'bulletedList']
            })
            .then(editor => {
                const data = editor.getData();
            })
            .catch(error => {
                console.error(error);
            });
    </script>
</body>
</html>
```
I created the custom layout because I wanted to have some paged with sidebar.

My worry is that, the wire:click event is not getting firering nothing in the network tab, neither in the console.
I have been stroogle with this for days already.
I even reinstall livewire using this: `php artisan jetstream:install livewire` then I ran `npm install` and `npm run dev` 
I also created a component using livewire `php artisan make:livewire test` itself and try to add a wire:click even but it still not get fired.

please I need some help.

**Comment by denizozturk** [comment]
Is there an error in the console tab? Can you type `dd('Test')` inside the mount function (to the top) in the livewire component and confirm if it works? I want to make sure that livewire is initializing correctly

**Comment by jonneroelofs** [comment]
Looks like you created a blade component and not a livewire component? Try php artisan make:livewire AddSection.

**Comment by joshhanley** [comment, marked as answer]
@kamdjouduplex yep @jonneroelofs is correct, you have created a Blade component, not a Livewire component.

To make a Livewire component you should have run

```bash
php artisan livewire:make AddSection
```

This will create two files, one in `resources/views/livewire/add-section.blade.php` and the other in `app/Http/Livewire/AddSection.php`.

Then in your `parent.blade.php`, you should add the Livewire component by using the Livewire tags like

```blade
<x-base-layout>
    <div class="max-w-7xl mx-auto bg-white">
        <div class="p-8">
            <h3 class="font-bold text-2xl">Editer le programme</h3>
        </div>
        <hr>
        <div class="p-8" id="coursSectionWraper">
            <livewire:add-section /> {{-- Notice this is different for Livewire --}}
        </div>
    </div>
</x-base-layout>
```

I strongly suggest you read through the Livewire documentation here, as it goes through all of that https://laravel-livewire.com/docs/2.x/quickstart

Hope this helps!

**Reply by kamdjouduplex** [reply]
@joshhanley Wooow!

It works,
`<livewire:add-section /> {{-- Notice this is different for Livewire --}}`
This was the proper way.

Just to clarify, what do we do with blade component then if livewire does not work in?

Thank you man.

**Reply by joshhanley** [reply]
@kamdjouduplex no worries 🙂

So you can still use blade components too, but if you have a `wire:click` (or other Livewire features) in a blade component, then that blade component must be wrapped in a Livewire component for them to work properly.

I would actually recommend using blade components as much as you can, but you need to make sure that you have a Livewire component wrapped around them all, so they work correctly.

Hope this helps!
